### PR TITLE
[PORT] New sounds when interacting with stuff

### PR DESCRIPTION
**Pull request by Hardly3D**
## About The Pull Request
This PR brings us closer to ~~Baystation~~ Shiptest by porting new sounds from Shiptest, which also just happens to be a bunch of stuff ported from other codebases as well.

This PR's changes includes:
- All computers now make clicking noises when interacted with, DNA scanners excluded so it isn't spammy.
- Sleepers now make sounds when interacted with as well as hypospray sounds when injecting someone with stuff.
- Buttons, APCs, fire alarms, vending machines and conveyors now makes a sound when interacted with.
- Hypospray has a new sound variant (Taken from Shiptest).
- Health and genetic analyzers makes a beep when analyzing.
- Mining scanners now have a speaker option that plays a sound when pulsed (On by default for manual ones, off for automatic ones so it doesn't get annoying).
- Beaker splashes, gun handling and looping sounds not included.

Ported from
https://github.com/shiptest-ss13/Shiptest/pull/130
https://github.com/shiptest-ss13/Shiptest/pull/484

## Why It's Good For The Game

Immersion, plannedemic, as well as more sounds are great for the game.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
Don't mind the crap quality and framerate, I don't even know how to set up OBS

Sleeper

https://user-images.githubusercontent.com/66234359/160212826-0afd9560-8771-46cb-be5c-2e32a9a8cf1f.mp4

Vending machine

https://user-images.githubusercontent.com/66234359/160213159-4417212a-af0a-4a5c-924b-5de200b75c7e.mp4

Buttons, fire alarms and conveyors

https://user-images.githubusercontent.com/66234359/160213854-a18e8586-0977-4076-abae-8c298937097c.mp4

APCs and Air Alarm

https://user-images.githubusercontent.com/66234359/160220362-5f6b6bd5-dd52-4c35-a4fb-497e127e450e.mp4

Computer

https://user-images.githubusercontent.com/66234359/160220380-4956b55b-1b22-484f-a9d8-d5664ed2f924.mp4

Analyzer

https://user-images.githubusercontent.com/66234359/160220428-a5ab9fec-b604-44a7-a4af-498438825dec.mp4

</details>

## Changelog
:cl: Hardly, Retlaw34, comma, mattroks101, metalgearsloth, tichys
add: New sounds for buttons, conveyors, fire alarms, health analyzers and mining scanners.
add: Sleepers now make a noise when used.
add: All computers now make clicking noises when interacted with.
add: New sound variant for hypospray.
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
Adding noise to dna scanner could give medical decent ambience tbh

**Comment by Hardly3D** [comment]
> Adding noise to dna scanner could give medical decent ambience tbh

It does sound like a good idea at first and I even gave that a try but it got spammy and annoying rather fast, hence why I excluded it from making any sounds.

**Comment by Kapu1178** [comment]
> Adding noise to dna scanner could give medical decent ambience tbh

ambient computers fucked up sound on shiptest

**Comment by Fronsis** [comment]
This would be nice alongside the other port (no idea what happened to the pr, but it's there) that ports the sounds of picking up different items that TG and the other branches have

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -576,5 +582,14 @@ Class Procs:
...
 	AM.pixel_y = -8 + (round( . / 3)*8)
 
+/obj/machinery/proc/play_click_sound(var/custom_clicksound)
+	if((custom_clicksound||clicksound) && world.time > next_clicksound)
+		next_clicksound = world.time + CLICKSOUND_INTERVAL
+		if(custom_clicksound)
+			playsound(src, custom_clicksound, clickvol)
+		else if(clicksound)
+			playsound(src, clicksound, clickvol)
+	return
```
Comment (ike709) [inline code/game/machinery/_machinery.dm:592]:
```suggestion
```

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -576,5 +582,14 @@ Class Procs:
 	AM.pixel_x = -8 + ((.%3)*8)
 	AM.pixel_y = -8 + (round( . / 3)*8)
 
+/obj/machinery/proc/play_click_sound(var/custom_clicksound)
+	if((custom_clicksound||clicksound) && world.time > next_clicksound)
+		next_clicksound = world.time + CLICKSOUND_INTERVAL
+		if(custom_clicksound)
+			playsound(src, custom_clicksound, clickvol)
+		else if(clicksound)
```
Comment (ike709) [inline code/game/machinery/_machinery.dm:590]:
```suggestion
		else
```
You already know one or the other has to be true.

File: code/game/machinery/_machinery.dm [hunk 3]
```diff
@@ -320,6 +324,8 @@ Class Procs:
 
 /obj/machinery/ui_act(action, params)
 	add_fingerprint(usr)
+	if(isliving(usr) && in_range(src, usr))
```
Comment (ike709) [inline code/game/machinery/_machinery.dm:327]: Is there not a range check anywhere before this?

File: code/game/sound.dm [hunk 4]
```diff
@@ -296,6 +296,14 @@ distance_multiplier - Can be used to multiply the distance at which the sound is
...
+			if("keystroke")
+				soundin = pick('sound/machines/keyboard/keypress1.ogg','sound/machines/keyboard/keypress2.ogg','sound/machines/keyboard/keypress3.ogg','sound/machines/keyboard/keypress4.ogg')
+			if("keyboard")
+				soundin = pick('sound/machines/keyboard/keystroke1.ogg','sound/machines/keyboard/keystroke2.ogg','sound/machines/keyboard/keystroke3.ogg','sound/machines/keyboard/keystroke4.ogg')
+			if("button")
```
Comment (ike709) [inline code/game/sound.dm:303]: I kind of hate that these aren't all defines but I guess it's out of scope for this PR.

**Review by ike709 — CHANGES_REQUESTED** [review]
Basically fine, just minor nitpicks. Also sounds good.

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (ike709) [inline code/game/machinery/_machinery.dm]:
```suggestion
/obj/machinery/proc/play_click_sound(var/custom_clicksound)
	if((custom_clicksound ||= clicksound) && world.time > next_clicksound)
		next_clicksound = world.time + CLICKSOUND_INTERVAL
		playsound(src, custom_clicksound, clickvol)
```

Comment (ike709) [reply code/game/machinery/_machinery.dm]: Should work identically.

File: code/modules/mining/equipment/mineral_scanner.dm [hunk 5]
```diff
@@ -61,6 +62,7 @@
 		current_cooldown = world.time + cooldown
 		var/turf/t = get_turf(src)
 		mineral_scan_pulse(t, range)
+		playsound(src, 'sound/effects/ping.ogg', 20)
```
Comment (Ivniinvi) [inline code/modules/mining/equipment/mineral_scanner.dm]: Please add a toggle with alt click or something, this gets super annoying when it's every two seconds for an hour and a half.

Comment (Hardly3D) [reply code/modules/mining/equipment/mineral_scanner.dm]: Done.

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (Hardly3D) [reply code/game/machinery/_machinery.dm]: Tested locally, seems to work well.

File: code/game/machinery/_machinery.dm (hunk 3 above)
Comment (Hardly3D) [reply code/game/machinery/_machinery.dm:327]: As far as I know, nope. The range check is there so it doesn't do clicking sounds while staying out of the range with the machine/computer that is being interacted with.

**Review by ike709 — APPROVED** [review]

**Pull request merged**